opensrp-opd/src/main/java/org/smartregister/opd/utils/JsonFormUtils.java

```diff
@@ -0,0 +1,1031 @@
+            List<FormLocation> upToFacilitiesWithOther = LocationHelper.getInstance().generateLocationHierarchyTree(true, healthFacilities);
+            List<FormLocation> entireTree = LocationHelper.getInstance().generateLocationHierarchyTree(true, allLevels);
+
+            String defaultLocationString = AssetHandler.javaToJsonString(defaultLocation, new TypeToken<List<String>>() {
+            }.getType());
+
+            String defaultFacilityString = AssetHandler.javaToJsonString(defaultFacility, new TypeToken<List<String>>() {
+            }.getType());
+
+            String upToFacilitiesString = AssetHandler.javaToJsonString(upToFacilities, new TypeToken<List<FormLocation>>() {
+            }.getType());
+
+            String upToFacilitiesWithOtherString = AssetHandler.javaToJsonString(upToFacilitiesWithOther, new TypeToken<List<FormLocation>>() {
+            }.getType());
+
+            String entireTreeString = AssetHandler.javaToJsonString(entireTree, new TypeToken<List<FormLocation>>() {
+            }.getType());
+
+            updateLocationTree(widgetKey, locationHierarchy, questions, defaultLocationString, defaultFacilityString, upToFacilitiesString, upToFacilitiesWithOtherString, entireTreeString);
+
+            //To Do Refactor to remove dependency on hardocded keys
+            for (int i = 0; i < questions.length(); i++) {
+                if (questions.getJSONObject(i).getString("key").equals("Home_Facility")) {
+                    if (StringUtils.isNotBlank(upToFacilitiesString)) {
+                        questions.getJSONObject(i).put("tree", new JSONArray(upToFacilitiesString));
+                    }
+                    if (StringUtils.isNotBlank(defaultFacilityString)) {
+                        questions.getJSONObject(i).put("default", defaultFacilityString);
+                    }
+                } else if (questions.getJSONObject(i).getString("key").equals("Birth_Facility_Name")) {
+                    if (StringUtils.isNotBlank(upToFacilitiesWithOtherString)) {
+                        questions.getJSONObject(i).put("tree", new JSONArray(upToFacilitiesWithOtherString));
+                    }
+                    if (StringUtils.isNotBlank(defaultFacilityString)) {
+                        questions.getJSONObject(i).put("default", defaultFacilityString);
+                    }
+                } else if (questions.getJSONObject(i).getString("key").equals("Residential_Area")) {
+                    if (StringUtils.isNotBlank(entireTreeString)) {
+                        questions.getJSONObject(i).put("tree", new JSONArray(entireTreeString));
+                    }
+                    if (StringUtils.isNotBlank(defaultLocationString)) {
+                        questions.getJSONObject(i).put("default", defaultLocationString);
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            Timber.e(e, "JsonFormUtils --> addRegLocHierarchyQuestions");
```

Comment:
no need since the whichever exception it throws will extend `Exception` so the class name will be given e.g `java.lang.IllegalArgumentException` , i think we usually specify a specific exception when we want do another thing when its thrown e.g either retry or something....or What do you think?